frontend/src/components/WildcardSelect.vue

```diff
@@ -0,0 +1,221 @@
+      v-model="wildcardSelectedValue"
+      :error-messages="getErrorMessages('wildcardSelectedValue')"
+      @input="onInput"
+      @blur="$v.wildcardSelectedValue.$touch()"
+      :hint="wildcardSelectHint"
+      persistent-hint
+    >
+      <template v-slot:item="{ item }">
+        <v-list-item-content>
+          <v-list-item-title>
+            <span v-if="item.startsWithWildcard">&lt;prefix&gt;</span>
+            <span>{{item.text}}</span>
+            <span v-if="item.endsWithWildcard">&lt;suffix&gt;</span>
+          </v-list-item-title>
+        </v-list-item-content>
+      </template>
+    </v-select>
+    <v-text-field
+      v-if="wildcardSelectedValue.isWildcard"
+      :class="textFieldClass"
+      color="cyan darken-2"
+      :label="wildcardTextFieldLabel"
+      v-model="wildcardVariablePart"
+      :error-messages="getErrorMessages('wildcardVariablePart')"
+      @input="onInput"
+      @blur="$v.wildcardVariablePart.$touch()"
+      ></v-text-field>
+  </div>
+</template>
+
+<script>
+import startsWith from 'lodash/startsWith'
+import endsWith from 'lodash/endsWith'
+import map from 'lodash/map'
+import trim from 'lodash/trim'
+import filter from 'lodash/filter'
+import head from 'lodash/head'
+import { getValidationErrors } from '@/utils'
+import { required, requiredIf } from 'vuelidate/lib/validators'
+
+const validations = {
+  wildcardVariablePart: {
+    required: requiredIf(function () {
+      return this.wildcardSelectedValue.isWildcard
+    })
+  },
+  wildcardSelectedValue: {
+    required
+  }
+}
+
+export default {
+  name: 'wildcard-select',
+  props: {
+    wildcardSelectItems: {
+      type: Array
+    },
+    wildcardSelectLabel: {
+      type: String
+    },
+    value: {
+      type: String
+    }
+  },
+  data () {
+    return {
+      wildcardVariablePart: undefined,
+      wildcardSelectedValue: {},
+      valid: undefined
+    }
+  },
+  computed: {
+    validationErrors () {
+      return {
+        wildcardVariablePart: {
+          required: `${this.wildcardTextFieldLabel} is required`
+        },
+        wildcardSelectedValue: {
+          required: `${this.wildcardSelectLabel} is required`
+        }
+      }
+    },
+    textFieldClass () {
+      return this.wildcardSelectedValue.startsWithWildcard ? 'textFieldStartClass' : 'textFieldEndClass'
+    },
+    wildcardSelectItemObjects () {
+      return map(this.wildcardSelectItems, item => {
+        let startsWithWildcard = false
+        let endsWithWildcard = false
+
+        let selectedValue = item
+        if (startsWith(item, '*')) {
+          startsWithWildcard = true
+        }
+        if (endsWith(item, '*')) {
+          endsWithWildcard = true
+        }
+        selectedValue = trim(item, '*')
+
+        return {
+          text: selectedValue,
+          value: selectedValue,
+          startsWithWildcard,
+          endsWithWildcard,
+          isWildcard: startsWithWildcard || endsWithWildcard
+        }
```

Comment:
why not building the regexp pattern already here. Something like
```js
        const trimmedValue = trim(item, '*')
        let pattern
        if (startsWith(item, '*')) {
          startsWithWildcard = true
          // TODO pattern = '^.+' + trimmedValue + '$'
        } else if (endsWith(item, '*')) {
          endsWithWildcard = true
          // TODO pattern = ...
        } else {
          // TODO pattern = ...
        }

        return {
          text: trimmedValue,
          value: trimmedValue,
          pattern,
          startsWithWildcard,
          endsWithWildcard,
          isWildcard: startsWithWildcard || endsWithWildcard
        }
```

Reply:
yes, I actually thought about this and had in that way before. did some refactoring and ended up not having the regex as part of the item object. I have no real opinion, if you like it better that way, I can change it.

Reply:
If you trim the value in one method and in the other you take this (trimmed) value and build the pattern for the regexp, it makes it harder to understand / follow. So for me it was not clear on first sight that `item.value` was actually the trimmed value.
This makes way makes it more clearer as it is done in one place.